gap/congruences/cong.gi

```diff
@@ -390,17 +424,69 @@ function(class, elm)
   return EquivalenceClassOfElementNC(cong, Representative(class) * elm);
 end);
 
+SEMIGROUPS.GenericCongLookup := function(cong)
+  local S, lookup, class, nr, elm;
+  S := Range(cong);
+  if not IsFinite(S) then
+    ErrorNoReturn("Semigroups: SEMIGROUPS.GenericCongLookup: usage,\n",
+                  "<cong> must be over a finite semigroup,");
+  fi;
+  lookup := [1 .. Size(S)];
+  for class in NonTrivialEquivalenceClasses(cong) do
+    nr := PositionCanonical(S, Representative(class));
```

Comment:
Okay, I've adapted this bit.  If the semigroup is infinite, `EquivalenceRelationLookup` can never have a sensible answer, so I'm sticking with an error message (but quoting `EquivalenceRelationLookup` instead of `SEMIGROUPS.GenericCongLookup).  If it's finite but non-enumerable then we go to `TryNextMethod()`